app/controllers/step_orderable_elements_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class StepOrderableElementsController < ApplicationController
+  before_action :load_vars
+  before_action :check_manage_permissions
+
+  def update
+    @step.with_lock do
+      element_to_swap = @step.step_orderable_elements.find_by(id: params[:new_position])
```

Comment:
I think we do not swap elements. We just move one element to the new position and all elements between old and new positions must be moved up/down depending on direction.